helm/hpcc/templates/esp.yaml

```diff
@@ -73,6 +73,12 @@ data:
 {{- if not .disabled -}}
 {{- $env := concat ($.Values.global.env | default list) (.env | default list) -}}
 {{- $application := .application | default "eclwatch" -}}
+{{- if not (hasKey . "service") -}}
+{{- $_ := set . "service" (dict "port" (.port | default 0) "servicePort" (.servicePort | default 0) "public" (.public | default false) ) -}}
+{{- $_ := unset . "port" -}}
+{{- $_ := unset . "servicePort" -}}
+{{- $_ := unset . "public" -}}
```

Comment:
is above for backward compatibility, for old value yamls that don't define the new .service.servicePort etc. spec?
If so, need similar for Sasha and Roxie to support their old format?

Reply:
Yes. Not sure whether it's a good idea or not, or whether should just require people to change. Given the schema changes I suspect this code may be moot anyway.

Reply:
yeah, given the schema, I think no point, though disruptive to users potentially.

If this code is removed though, I think there is other code that node checks .service.X - that will need to be careful to check if key "service" exists first.